src/libjasper/jpc/jpc_t1enc.c

```diff
@@ -299,6 +296,13 @@ assert(jas_stream_tell(cblk->stream) == jas_stream_getrwcount(cblk->stream));
 			break;
 		}
 
+		if (ret) {
+			if (bout) {
+				jpc_bitstream_close(bout);
+			}
+			return -1;
+		}
+
```

Comment:
This piece of code is not related to fixing warnings. However, it is good to have it.

Reply:
True, it's part of commit b71f9751175937459905c2360220a18f4ffcb80e which fixes error handling in jpc_enc_enccblk().